apps/common/js/components/leap-hands-view.jsx

```diff
@@ -11,26 +12,28 @@ const SKIN_COLOR = 0x93603F;
 export default class LeapHandsView extends React.Component {
   componentDidMount() {
     const { handsOpacity } = this.props;
-    const renderer = new THREE.WebGLRenderer({alpha: true});
-    renderer.setClearColor(0x000000, 0);
-    renderer.setSize(this.width, this.height);
-    this.refs.container.appendChild(renderer.domElement);
-    const threeData = this.initScene();
+    this.scene = new THREE.Scene();
+    this.renderer = new THREE.WebGLRenderer({ alpha: true });
+    this.refs.container.appendChild(this.renderer.domElement);
+    this.initScene();
+
     leapController.use('riggedHand', {
-      renderer,
-      parent: threeData.scene,
-      scene: threeData.scene,
-      camera: threeData.camera,
+      renderer: this.renderer,
+      parent: this.scene,
+      scene: this.scene,
+      camera: this.camera,
       materialOptions: {
         opacity: handsOpacity
       }
     });
-    leapController.on('riggedHand.meshAdded', function(handMesh){
+    leapController.on('riggedHand.meshAdded', function (handMesh) {
       handMesh.material.color.setHex(SKIN_COLOR);
       handMesh.material.emissive.setHex(0x000000);
       handMesh.material.ambient.setHex(SKIN_COLOR);
     });
-
+  }
+  componentWillReceiveProps(newProps) {
+    this.resize();
```

Comment:
This call makes perfect sense here since we have `width` and `height` in properties only. However the funny thing is that we never change size of this component using its properties - we always keep width and height equal to 100% and modify parent's dimensions using CSS. Also, it's probably impossible to update width and height of this component using properties as `shouldComponentUpdate` always returns `false` ([docs](https://facebook.github.io/react/docs/component-specs.html#updating-shouldcomponentupdate)).

But it still works fine somehow. `resize()` gets called anyway, as probably React starts the whole update process when we change pulldowns and re-arrange views. So, in other words - this call is reasonable, but I feel it works in our case by accident. 😉  Or theoretically it shouldn't / doesn't have to.

It's more code and less pretty, but I guess chain of the explicit `.resize()` calls in parent components is what we should use in our case to be clearer and more bulletproof. Or use props to change dimensions of this component and then leave this code above.

I know it works in practice, but I hope you know what I mean. And that's probably a nice way to understand some React quirks. :wink: